diff --git a/dist/components/portal/portal.js b/dist/components/portal/portal.js
index b84fcf7fd1de1768907dbc57fb3c8dd3e4fc0fc8..369e517941fc44c98f0260f131697b0f9d3602d3 100644
--- a/dist/components/portal/portal.js
+++ b/dist/components/portal/portal.js
@@ -9,16 +9,16 @@ var Portal = (props) => {
 	const { children, disabled } = props;
 	const [container, setContainer] = useState(props.container?.current);
 	const isServer = useSyncExternalStore(subscribe, () => false, () => true);
-	const { getRootNode } = useEnvironmentContext();
+	const { getRootNode, getPortalNode } = useEnvironmentContext();
 	useEffect(() => {
 		setContainer(() => props.container?.current);
 	}, [props.container]);
 	if (isServer || disabled) return /* @__PURE__ */ jsx(Fragment$1, { children });
-	const mountNode = container ?? getPortalNode(getRootNode);
+	const mountNode = container ?? getPortalNode?.() ?? getContainer(getRootNode);
 	return /* @__PURE__ */ jsx(Fragment$1, { children: Children.map(children, (child) => createPortal(child, mountNode)) });
 };
-var getPortalNode = (cb) => {
-	const node = cb?.();
+var getContainer = (getRootNode) => {
+	const node = getRootNode?.();
 	const rootNode = node.getRootNode();
 	if (isShadowRoot(rootNode)) return rootNode;
 	return getDocument(node).body;
diff --git a/dist/providers/environment/environment-provider.d.ts b/dist/providers/environment/environment-provider.d.ts
index 335d2668b745d6a6765024b0602507408f0c08fa..50084351aea837c54b427e7a0f717b56d10c490d 100644
--- a/dist/providers/environment/environment-provider.d.ts
+++ b/dist/providers/environment/environment-provider.d.ts
@@ -3,5 +3,6 @@ import { RootNode } from './use-environment-context';
 export interface EnvironmentProviderProps {
     children?: ReactNode | undefined;
     value?: RootNode | (() => RootNode) | undefined;
+    portalNode?: Node | undefined;
 }
 export declare const EnvironmentProvider: (props: EnvironmentProviderProps) => import("react/jsx-runtime").JSX.Element;
diff --git a/dist/providers/environment/environment-provider.js b/dist/providers/environment/environment-provider.js
index c92c16a73cbdcea790b30d10397b7247285d8331..7eab5fd31c01fa2ad02553b8638b1ffa59d8f222 100644
--- a/dist/providers/environment/environment-provider.js
+++ b/dist/providers/environment/environment-provider.js
@@ -6,7 +6,7 @@ import { jsx, jsxs } from "react/jsx-runtime";
 import { getDocument, getWindow } from "@zag-js/dom-query";
 //#region src/providers/environment/environment-provider.tsx
 var EnvironmentProvider = (props) => {
-	const { value, children } = props;
+	const { value, portalNode, children } = props;
 	const [spanRef, setSpanRef] = useState();
 	const getRootNode = useMemo(() => {
 		return () => runIfFn(value) ?? spanRef?.getRootNode() ?? document;
@@ -14,6 +14,7 @@ var EnvironmentProvider = (props) => {
 	return /* @__PURE__ */ jsxs(EnvironmentContextProvider, {
 		value: useMemo(() => ({
 			getRootNode,
+			getPortalNode: () => portalNode || undefined,
 			getWindow: () => getWindow(getRootNode()),
 			getDocument: () => getDocument(getRootNode())
 		}), [getRootNode]),
diff --git a/dist/providers/environment/use-environment-context.d.ts b/dist/providers/environment/use-environment-context.d.ts
index 67b730775a75bbab8d8f8d16699ace93d3378c5e..c745b7b67f9e8120add173cda5597c4bf206b220 100644
--- a/dist/providers/environment/use-environment-context.d.ts
+++ b/dist/providers/environment/use-environment-context.d.ts
@@ -6,9 +6,14 @@ export interface UseEnvironmentContext {
      * This is used to determine the window and document objects.
      */
     getRootNode(): RootNode;
+    /**
+     * Default target for portal content.
+     */
+    getPortalNode(): Node | undefined;
     /**
      * The document context for the root node.
      */
+    
     getDocument(): Document;
     /**
      * The window context for the root node.
diff --git a/dist/providers/environment/use-environment-context.js b/dist/providers/environment/use-environment-context.js
index 474fdd6428cb8b72a0d2d8cf65b11668885df5d0..1076946f6813da2b3900bd142b230eec9dd7cafe 100644
--- a/dist/providers/environment/use-environment-context.js
+++ b/dist/providers/environment/use-environment-context.js
@@ -9,7 +9,8 @@ var [EnvironmentContextProvider, useEnvironmentContext] = createContext({
 	defaultValue: {
 		getRootNode: () => document,
 		getDocument: () => document,
-		getWindow: () => window
+		getWindow: () => window,
+		getPortalNode: () => document,
 	}
 });
 //#endregion
