diff --git a/dist/chunk-R5W6LHQW.mjs b/dist/chunk-R5W6LHQW.mjs
index 75c349637d59f487c75f477f0cea9ad7b5e45fff..8a41cf24fc40770e30553fe572cb722edda6b4c8 100644
--- a/dist/chunk-R5W6LHQW.mjs
+++ b/dist/chunk-R5W6LHQW.mjs
@@ -60,7 +60,7 @@ function isValidEvent(event, ref) {
     if (!doc.contains(target))
       return false;
   }
-  return !((_a = ref.current) == null ? void 0 : _a.contains(target));
+  return !((_a = ref.current) == null ? void 0 : event.composedPath().includes(_a));
 }
 
 export {
