diff --git a/dist/index.mjs b/dist/index.mjs
index 2d6e217a22b70ae9e87d0e87dd57c99a1fea48fa..831d92871f165986747f69a7738447501ca7113c 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -57,7 +57,7 @@ function isValidEvent(event, ref) {
     if (!doc.contains(target))
       return false;
   }
-  return !((_a = ref.current) == null ? void 0 : _a.contains(target));
+  return !((_a = ref.current) == null ? void 0 : event.composedPath().includes(_a));
 }
 function getOwnerDocument(node) {
   var _a;
