diff --git a/dist/index.js b/dist/index.js
index 7ea915636cd2285310a3848e67533fefa512e9d9..d1d790bede35cd509020ec44d1b72487ecba5d5b 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -1732,7 +1732,7 @@ async function getOptimizeDepsIncludes(rootDir) {
     const serviceModule = pkg2.servicesModuleId;
     return serviceModule ? [serviceModule] : [];
   });
-  trailsModules.push(`${RuntimeSupport3.RUNTIME_PACKAGE_NAME}/**`);
+  //trailsModules.push(`${RuntimeSupport3.RUNTIME_PACKAGE_NAME}/**`);
   isDebug5 && debug6("Optimizing additional modules %O", trailsModules);
   return trailsModules;
 }
