diff --git a/index.cjs b/index.cjs
index 7d3182ffc4421a773202514222ebe22a963cd840..6a2875503557f8f93cd52608097c7b9e71b8639d 100644
--- a/index.cjs
+++ b/index.cjs
@@ -62,7 +62,7 @@ var react = (_options) => {
       async transform(code, _id, transformOptions) {
         const id = _id.split("?")[0];
         const refresh = !(transformOptions == null ? void 0 : transformOptions.ssr) && !process.env.TEST;
-        const result = await transformWithOptions(id, code, "es2020", options, {
+        const result = await transformWithOptions(id, code, "es2022", options, {
           refresh,
           development: true,
           runtime: "automatic",
diff --git a/index.mjs b/index.mjs
index 2fe63c7dc3d5fe27ae6947cac737a27c80c37303..d5c35d0ad40a1fc2d4471dd6e4e2dcef41e8a10d 100644
--- a/index.mjs
+++ b/index.mjs
@@ -63,7 +63,7 @@ var react = (_options) => {
       async transform(code, _id, transformOptions) {
         const id = _id.split("?")[0];
         const refresh = !(transformOptions == null ? void 0 : transformOptions.ssr) && !process.env.TEST;
-        const result = await transformWithOptions(id, code, "es2020", options, {
+        const result = await transformWithOptions(id, code, "es2022", options, {
           refresh,
           development: true,
           runtime: "automatic",