diff --git a/dist/tooltip.dom.mjs b/dist/tooltip.dom.mjs
index 2942e3e5bdbacb699015a0bf1cccfb6778a0115d..b20b4bd85b54674d78ccdc735a1e469beab01291 100644
--- a/dist/tooltip.dom.mjs
+++ b/dist/tooltip.dom.mjs
@@ -13,9 +13,12 @@ var getTriggerEl = (scope) => scope.getById(getTriggerId(scope));
 var getContentEl = (scope) => scope.getById(getContentId(scope));
 var getPositionerEl = (scope) => scope.getById(getPositionerId(scope));
 var getArrowEl = (scope) => scope.getById(getArrowId(scope));
-var getTriggerEls = (scope) => queryAll(scope.getDoc(), `[data-scope="tooltip"][data-part="trigger"][data-ownedby="${scope.id}"]`);
+var getTriggerEls = (scope) => queryAll(scope.getRootNode(), `[data-scope="tooltip"][data-part="trigger"][data-ownedby="${scope.id}"]`);
 var getActiveTriggerEl = (scope, value) => {
-  return value == null ? getTriggerEls(scope)[0] : scope.getById(getTriggerId(scope, value));
+  if (value == null) {
+    return getTriggerEl(scope) ?? getTriggerEls(scope)[0]
+  }
+  return scope.getById(getTriggerId(scope, value))
 };
 export {
   getActiveTriggerEl,
