diff --git a/dist/index.mjs b/dist/index.mjs
index 78d84961d8c73ffa8269e5ce9411e8c6ba855b73..6e0501c7f4d595fa0d8662a6a1ac07c82735cff0 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -67,13 +67,19 @@ function setupGlobalFocusEvents(root) {
   }
   const win = getWindow(root);
   const doc = getDocument(root);
+  
   let focus = win.HTMLElement.prototype.focus;
-  win.HTMLElement.prototype.focus = function() {
+  function patchedFocus() {
     currentModality = "virtual";
     triggerChangeHandlers("virtual", null);
     hasEventBeforeFocus = true;
     focus.apply(this, arguments);
-  };
+  }
+  Object.defineProperty(win.HTMLElement.prototype, "focus", {
+    configurable: true,
+    value: patchedFocus
+  });
+
   doc.addEventListener("keydown", handleKeyboardEvent, true);
   doc.addEventListener("keyup", handleKeyboardEvent, true);
   doc.addEventListener("click", handleClickEvent, true);
