diff --git a/dist/index.mjs b/dist/index.mjs
index 27309b0b6538702bb6bc243cd6e6135aa4030eb9..8dd4c297593df08a0a8f82841c4b08dbeec45983 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -374,7 +374,9 @@ var groupMachine = createMachine({
     },
     effects: {
       subscribeToStore({ context, prop }) {
-        return prop("store").subscribe((toast) => {
+        const store = prop("store");
+        context.set("toasts", store.getVisibleToasts());
+        return store.subscribe((toast) => {
           if (toast.dismiss) {
             context.set("toasts", (prev) => prev.filter((t) => t.id !== toast.id));
             return;
