diff --git a/dist/popover.dom.mjs b/dist/popover.dom.mjs
index b46929d79b9d2dbb8cf69b5cbe2a63ce85dad042..3aedc99fc1c17014fc0371a1ec264355591994f5 100644
--- a/dist/popover.dom.mjs
+++ b/dist/popover.dom.mjs
@@ -14,9 +14,13 @@ var getTitleId = (scope) => scope.ids?.title ?? `popover:${scope.id}:title`;
 var getDescriptionId = (scope) => scope.ids?.description ?? `popover:${scope.id}:desc`;
 var getCloseTriggerId = (scope) => scope.ids?.closeTrigger ?? `popover:${scope.id}:close`;
 var getAnchorEl = (scope) => scope.getById(getAnchorId(scope));
-var getTriggerEls = (scope) => queryAll(scope.getDoc(), `[data-scope="popover"][data-part="trigger"][data-ownedby="${scope.id}"]`);
+var getTriggerEl = (scope) => scope.getById(getTriggerId(scope));
+var getTriggerEls = (scope) => queryAll(scope.getRootNode(), `[data-scope="popover"][data-part="trigger"][data-ownedby="${scope.id}"]`);
 var getActiveTriggerEl = (scope, value) => {
-  return value == null ? getTriggerEls(scope)[0] : scope.getById(getTriggerId(scope, value));
+  if (value == null) {
+    return getTriggerEl(scope) ?? getTriggerEls(scope)[0]
+  }
+  return scope.getById(getTriggerId(scope, value));
 };
 var getContentEl = (scope) => scope.getById(getContentId(scope));
 var getPositionerEl = (scope) => scope.getById(getPositionerId(scope));
@@ -40,6 +44,7 @@ export {
   getPositionerId,
   getTitleEl,
   getTitleId,
+  getTriggerEl,
   getTriggerEls,
   getTriggerId
 };
diff --git a/dist/popover.machine.mjs b/dist/popover.machine.mjs
index c657381410fb44218105a9c550527852e92fd2d1..52444c1fbd0cad83f64c88fa6034ace692274c51 100644
--- a/dist/popover.machine.mjs
+++ b/dist/popover.machine.mjs
@@ -158,7 +158,7 @@ var machine = createMachine({
         return trackDismissableElement(getContentEl2, {
           type: "popover",
           pointerBlocking: prop("modal"),
-          exclude: dom.getTriggerEls(scope),
+          exclude: [dom.getTriggerEl(scope), ...dom.getTriggerEls(scope)].filter(Boolean),
           defer: true,
           onEscapeKeyDown(event) {
             prop("onEscapeKeyDown")?.(event);
