diff --git a/dist/drawer.dom.mjs b/dist/drawer.dom.mjs
index 19684cfa0d0dfd1284fc98ba1ed9308a66947f8c..740a82efa19186c7a056daf90d1ee4ce0f15a9db 100644
--- a/dist/drawer.dom.mjs
+++ b/dist/drawer.dom.mjs
@@ -12,7 +12,7 @@ var getTriggerId = (ctx, value) => {
   if (customId != null) return isFunction(customId) ? customId(value) : customId;
   return value ? `drawer:${ctx.id}:trigger:${value}` : `drawer:${ctx.id}:trigger`;
 };
-var getTriggerEls = (ctx) => queryAll(ctx.getDoc(), `[data-scope="drawer"][data-part="trigger"][data-ownedby="${ctx.id}"]`);
+var getTriggerEls = (ctx) => queryAll(ctx.ctx.getRootNode(), `[data-scope="drawer"][data-part="trigger"][data-ownedby="${ctx.id}"]`);
 var getActiveTriggerEl = (ctx, value) => {
   if (value == null) return getTriggerEl(ctx) ?? getTriggerEls(ctx)[0];
   return ctx.getById(getTriggerId(ctx, value));
diff --git a/dist/drawer.machine.mjs b/dist/drawer.machine.mjs
index 70d61b98762e3df178654b24ed54d158a7572690..bd672cb9e96a5e122256efb6a1ea2cf5a9b2dd50 100644
--- a/dist/drawer.machine.mjs
+++ b/dist/drawer.machine.mjs
@@ -599,7 +599,7 @@ var machine = createMachine({
           type: "drawer",
           defer: true,
           pointerBlocking: prop("modal"),
-          exclude: [dom.getTriggerEl(scope)],
+          exclude: [dom.getTriggerEl(scope), ...dom.getTriggerEls(scope)].filter(Boolean),
           onInteractOutside(event) {
             prop("onInteractOutside")?.(event);
             if (!prop("closeOnInteractOutside")) {
