diff --git a/dist/Select-1b9abbe3.cjs.dev.js b/dist/Select-1b9abbe3.cjs.dev.js
index d7ba84c3a2ee46188b6542ec1a33032ecb5e87ff..94e89a6c6d51561d89613e9bc3a3636d2559ac2d 100644
--- a/dist/Select-1b9abbe3.cjs.dev.js
+++ b/dist/Select-1b9abbe3.cjs.dev.js
@@ -1524,8 +1524,11 @@ var Select = /*#__PURE__*/function (_Component) {
       // close the menu if the user taps outside
       // we're checking on event.target here instead of event.currentTarget, because we want to assert information
       // on events on child elements, not the document (which we've attached this handler to).
-      if (_this.controlRef && !_this.controlRef.contains(event.target) && _this.menuListRef && !_this.menuListRef.contains(event.target)) {
-        _this.blurInput();
+      // ---
+      // PATCH: Fix event handling in shadow DOM + Touch input. Use Chrome Dev Tools with touch emulation and 
+      // attempt to use the input element + dropdown menu to verify this patch.
+      if (_this.controlRef && !event.composedPath().includes(_this.controlRef) && _this.menuListRef && !event.composedPath().includes(_this.menuListRef)) {
+         _this.blurInput();
       }
 
       // reset move vars
@@ -2157,7 +2160,8 @@ var Select = /*#__PURE__*/function (_Component) {
         'aria-labelledby': this.props['aria-labelledby'],
         'aria-required': required,
         role: 'combobox',
-        'aria-activedescendant': this.state.isAppleDevice ? undefined : this.state.focusedOptionId || ''
+        // PATCH: Firefox warns if this is the empty string
+        'aria-activedescendant': this.state.isAppleDevice ? undefined : this.state.focusedOptionId || undefined
       }, menuIsOpen && {
         'aria-controls': this.getElementId('listbox')
       }), !isSearchable && {
diff --git a/dist/Select-8685a2b1.cjs.prod.js b/dist/Select-8685a2b1.cjs.prod.js
index 528cc2efd0eed39a5d032a8183dbc6d9ca1b1e5a..d6bf0be316cc87453111bd9c5b534864613e04e7 100644
--- a/dist/Select-8685a2b1.cjs.prod.js
+++ b/dist/Select-8685a2b1.cjs.prod.js
@@ -1505,8 +1505,11 @@ var Select = /*#__PURE__*/function (_Component) {
       // close the menu if the user taps outside
       // we're checking on event.target here instead of event.currentTarget, because we want to assert information
       // on events on child elements, not the document (which we've attached this handler to).
-      if (_this.controlRef && !_this.controlRef.contains(event.target) && _this.menuListRef && !_this.menuListRef.contains(event.target)) {
-        _this.blurInput();
+      // ---
+      // PATCH: Fix event handling in shadow DOM + Touch input. Use Chrome Dev Tools with touch emulation and 
+      // attempt to use the input element + dropdown menu to verify this patch.
+      if (_this.controlRef && !event.composedPath().includes(_this.controlRef) && _this.menuListRef && !event.composedPath().includes(_this.menuListRef)) {
+         _this.blurInput();
       }
 
       // reset move vars
@@ -2138,7 +2141,8 @@ var Select = /*#__PURE__*/function (_Component) {
         'aria-labelledby': this.props['aria-labelledby'],
         'aria-required': required,
         role: 'combobox',
-        'aria-activedescendant': this.state.isAppleDevice ? undefined : this.state.focusedOptionId || ''
+        // PATCH: Firefox warns if this is the empty string
+        'aria-activedescendant': this.state.isAppleDevice ? undefined : this.state.focusedOptionId || undefined
       }, menuIsOpen && {
         'aria-controls': this.getElementId('listbox')
       }), !isSearchable && {
diff --git a/dist/Select-ef7c0426.esm.js b/dist/Select-ef7c0426.esm.js
index 1a01efb57d48e84665aca3117c9219d76525e654..56c8e19647fb3582da7d2ae620868e2dd1bb523f 100644
--- a/dist/Select-ef7c0426.esm.js
+++ b/dist/Select-ef7c0426.esm.js
@@ -1500,8 +1500,11 @@ var Select = /*#__PURE__*/function (_Component) {
       // close the menu if the user taps outside
       // we're checking on event.target here instead of event.currentTarget, because we want to assert information
       // on events on child elements, not the document (which we've attached this handler to).
-      if (_this.controlRef && !_this.controlRef.contains(event.target) && _this.menuListRef && !_this.menuListRef.contains(event.target)) {
-        _this.blurInput();
+      // ---
+      // PATCH: Fix event handling in shadow DOM + Touch input. Use Chrome Dev Tools with touch emulation and 
+      // attempt to use the input element + dropdown menu to verify this patch.
+      if (_this.controlRef && !event.composedPath().includes(_this.controlRef) && _this.menuListRef && !event.composedPath().includes(_this.menuListRef)) {
+         _this.blurInput();
       }
 
       // reset move vars
@@ -2133,7 +2136,8 @@ var Select = /*#__PURE__*/function (_Component) {
         'aria-labelledby': this.props['aria-labelledby'],
         'aria-required': required,
         role: 'combobox',
-        'aria-activedescendant': this.state.isAppleDevice ? undefined : this.state.focusedOptionId || ''
+        // PATCH: Firefox warns if this is the empty string
+        'aria-activedescendant': this.state.isAppleDevice ? undefined : this.state.focusedOptionId || undefined
       }, menuIsOpen && {
         'aria-controls': this.getElementId('listbox')
       }), !isSearchable && {
